pkg/api/v1/types.go

```diff
@@ -272,6 +274,12 @@ const (
 	// If there happens any error while trying to synchronize the VirtualMachineInstance with the Domain,
 	// this is reported as false.
 	VirtualMachineInstanceSynchronized VirtualMachineInstanceConditionType = "Synchronized"
+
+	// Reflects whether the QEMU guest agent is connected through the channel
+	VirtualMachineInstanceAgentConnected VirtualMachineInstanceConditionType = "AgentConnected"
+
+	// Reflects whether the QEMU guest agent is connected through the channel
+	VirtualMachineInstanceAgentChannelOpened VirtualMachineInstanceConditionType = "ChannelOpen"
```

Comment:
Has `VirtualMachineInstanceAgentChannelOpened` additional value compared to `VirtualMachineInstanceAgentConnected`?

Reply:
Ideally we can represent the agent with the three states of one condition, not sure if that's possible:

 * connected, add the condition and set to true
 * we see that it should be there but it is not connected, set to false. Here it is a little bit unclear to me what libvirt reports when. Is it possible to distinguish between a disconnected agent and no agent being discovered?
 * we see that the guest agent can't be there, so we don't add the condition.